common/reviews/api/gulp-core-build.api.ts

```diff
@@ -58,56 +59,57 @@ class GulpTask<TASK_CONFIG> implements IExecutable {
   public enabled: boolean;
   public execute(config: IBuildConfig): Promise<void>;
   // WARNING: The type "GulpProxy" needs to be exported by the package (e.g. added to index.ts)
-  public abstract executeTask(gulp: gulp.Gulp | GulpProxy, completeCallback?: (result?: Object) => void): Promise<Object> | NodeJS.ReadWriteStream | void;
+  public abstract executeTask(gulp: gulp.Gulp | GulpProxy,
+      completeCallback?: (error?: string | Error) => void): Promise<Object> | NodeJS.ReadWriteStream | void;
   public fileError(filePath: string, line: number, column: number, errorCode: string, message: string): void;
   public fileExists(localPath: string): boolean;
   public fileWarning(filePath: string, line: number, column: number, warningCode: string, message: string): void;
   public getCleanMatch(buildConfig: IBuildConfig, taskConfig: TASK_CONFIG = this.taskConfig): string[];
   public isEnabled(buildConfig: IBuildConfig): boolean;
-  protected loadSchema(): Object;
+  protected loadSchema(): Object | undefined;
   public log(message: string): void;
   public logError(message: string): void;
   public logVerbose(message: string): void;
   public logWarning(message: string): void;
   public mergeConfig(taskConfig: TASK_CONFIG): void;
   public name: string;
   public onRegister(): void;
-  public readJSONSync(localPath: string): Object;
+  public readJSONSync(localPath: string): Object | undefined;
   public replaceConfig(taskConfig: TASK_CONFIG): void;
   public resolvePath(localPath: string): string;
-  public readonly schema: Object;
+  public readonly schema: Object | undefined;
   public setConfig(taskConfig: TASK_CONFIG): void;
   public taskConfig: TASK_CONFIG;
 }
 
 // @public (undocumented)
 interface IBuildConfig {
-  args?: {
+  args: {
     [ name: string ]: string | boolean
   }
   buildErrorIconPath?: string;
   buildSuccessIconPath?: string;
-  distFolder?: string;
-  gulp?: GulpProxy | gulp.Gulp;
+  distFolder: string;
+  gulp: GulpProxy | gulp.Gulp;
   isRedundantBuild?: boolean;
   libAMDFolder?: string;
   libES6Folder?: string;
-  libFolder?: string;
+  libFolder: string;
   onTaskEnd?: (taskName: string, duration: number[], error?: any) => void;
   onTaskStart?: (taskName: string) => void;
-  packageFolder?: string;
-  production?: boolean;
+  packageFolder: string;
+  production: boolean;
   properties?: {
     [ key: string ]: any
   }
   relogIssues?: boolean;
-  rootPath?: string;
+  rootPath: string;
   shouldWarningsFailBuild?: boolean;
   showToast?: boolean;
-  srcFolder?: string;
-  tempFolder?: string;
+  srcFolder: string;
+  tempFolder: string;
   uniqueTasks?: IExecutable[];
-  verbose?: boolean;
```

Comment:
They're guaranteed to be defined. Otherwise a task that uses StrictNullChecks would have to check if it's defined before using it.

---
In reply to: [124909476](https://github.com/Microsoft/web-build-tools/pull/256#discussion_r124909476) [](ancestors = 124909476)